pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -110,6 +110,9 @@ type SensorSpec struct {
 	// RevisionHistoryLimit specifies how many old deployment revisions to retain
 	// +optional
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty" protobuf:"varint,7,opt,name=revisionHistoryLimit"`
+	// LoggingFields add additional key-value pairs when logging happens
+	// +optional
+	LoggingFields []LoggingField `json:"loggingFields" protobuf:"bytes,8,opt,name=loggingFields"`
```

Comment:
Is a `map[string]string` good enough?

use `rep` instead of `opt` in the `json` part

Reply:
Thanks for the feedback. I am new to go. I believe if I use `map[string]string`, I won't be able to define the fields (name and value)?

ok! I will change it! I am curious what is the difference between `opt` and `rep`?

Reply:
Thanks for the feedback. I am new to go. I believe if I use `map[string]string`, I won't be able to define the fields (name and value)?

ok! I will change it! I am curious what is the difference between `opt` and `rep`?

Reply:
`rep` means repeated fields in protobuf.


With `map[string]string`, you do
```yaml
  loggingFields:
    a: b
    e: f
```

Reply:
> `rep` means repeated fields in protobuf.
> 
> With `map[string]string`, you do
> 
> ```yaml
>   loggingFields:
>     a: b
>     e: f
> ```

oh I get what you mean now. I was initially thinking to make it look like
```
loggingFields:
- name: job_id
   value: 123
```
and you are suggesting to make it
```
loggingFields:
    job_id: abc
```
Your approach is better

Reply:
oh I get what you mean now. I was initially thinking to make it look like
```
loggingFields:
- name: job_id
  value: 123
```
and you are suggesting to make it
```
loggingFields:
    job_id: abc
```
Your approach is better